python_modules/dagster/dagster/_core/definitions/decorators/repository_decorator.py

```diff
@@ -52,6 +53,99 @@ def _flatten(items: Iterable[Union[T, List[T]]]) -> Iterator[T]:
             yield x
 
 
+def create_repository_data_from_fn(
```

Comment:
pulled more or less wholesale from the existing repo creation logic (so that it can be deferred later in LambdaRepositoryData)

can_defer_repository_data is new - for now i disabled that functionality for these